src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelValidationNode.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Captures the validation information for a particular model.
+    /// </summary>
+    public class ModelValidationNode
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="ModelValidationNode"/>.
+        /// </summary>
+        /// <param name="key">The key that will be used by the validation system to log <see cref="ModelState"/>
+        /// entries.</param>
+        /// <param name="modelMetadata">The <see cref="ModelMetadata"/> for the <paramref name="model"/>.</param>
+        /// <param name="model">The model object which is to be validated.</param>
+        public ModelValidationNode([NotNull] string key, [NotNull] ModelMetadata modelMetadata, object model)
+            : this (key, modelMetadata, model, new List<ModelValidationNode>())
+        {
+        }
+        /// <summary>
+        /// Creates a new instance of <see cref="ModelValidationNode"/>.
+        /// </summary>
+        /// <param name="key">The key that will be used by the validation system to add
+        /// <see cref="ModelStateDictionary"/> entries.</param>
+        /// <param name="modelMetadata">The <see cref="ModelMetadata"/> for the <paramref name="model"/>.</param>
+        /// <param name="model">The model object which will be validated.</param>
+        /// <param name="childNodes">A collection of child nodes.</param>
```

Comment:
Minor because I doubt we have scenarios where callers of this constructor care about the `IList` independently: Should this parameter be an `IEnumerable<ModelValidationNode>` which is _copied_ into a `List` the `ModelValidationNode` always constructs?